File: Backend/app/api/v1/health.py
from typing import Dict, Any
from fastapi import APIRouter
from app.core.config import settings
from app.services.gemini_service import GeminiService
from app.services.providers.sarvam_provider import SarvamProvider

router = APIRouter(tags=["Health & Status"])


@router.get("/health")
def health_check() -> Dict[str, Any]:
    """Returns overall health, database availability, and AI provider connectivity status."""
    sarvam = SarvamProvider()
    return {
        "status": "healthy",
        "project": settings.PROJECT_NAME,
        "environment": settings.ENVIRONMENT,
        "ai_providers": {
            "gemini": {
                "configured": bool(settings.GEMINI_API_KEY),
                "available": GeminiService.is_available(),
                "model": settings.GEMINI_MODEL,
            },
            "sarvam": {
                "configured": bool(settings.SARVAM_API_KEY),
                "available": sarvam.is_available(),
            },
            "azure_speech": {
                "configured": bool(settings.AZURE_SPEECH_KEY),
            },
        },
    }

File: FRONT-END/src/api/health.ts
import { apiClient } from './client';

export interface HealthBackendResponse {
  status: string;
  project: string;
  environment: string;
  ai_providers?: {
    gemini?: { configured: boolean; available: boolean; model: string };
    sarvam?: { configured: boolean; available: boolean };
    azure_speech?: { configured: boolean };
  };
}

export const healthApi = {
  async checkHealth(): Promise<HealthBackendResponse> {
    return apiClient<HealthBackendResponse>('/health', {
      method: 'GET',
      skipAuth: true,
    });
  },
};
